### PR TITLE
fixes #1180 - Reject app with existing id on POST to /v2/apps

### LINKED ISSUE #1180: Disallow apps the be overwritten with a POST
If a user POSTs an app with an existing id, the request should be rejected with status code 409 - Conflict.

### PR DESCRIPTION
**Pull request by drexin**

### CODE REVIEW

**Comment by ConnorDoyle** [comment]
Per the conversation in issue #1180, would you like to also modify the response JSON for PUT when creating an app resource?  Otherwise, let's make another issue so that the idea doesn't get lost when this PR closes that issue.

**Comment by drexin** [comment]
I was just about to do that ;-).

File: src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala [hunk 1]
```diff
@@ -91,7 +91,14 @@ class AppsResource @Inject() (
   def create(@Context req: HttpServletRequest, body: Array[Byte],
              @DefaultValue("false")@QueryParam("force") force: Boolean): Response = {
     val app = Json.parse(body).as[AppDefinition]
-    create(req, app, force)
+    service.getApp(app.id) match {
+      case None => create(req, app, force)
+      case _ => Response.status(409).entity(
+        Json.obj(
+          "message" -> "app with same id already exists"
```
Comment (ConnorDoyle) [inline src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala]: Let's output the conflicting app ID as part of the message.

```json
{
  "message": "An app with id [/some/test/app] already exists."
}
```

Comment (drexin) [inline src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala]: Yes, I will need to rebase with the other changes anyways, so I will just combine it with this change. Thanks!

**Comment by ConnorDoyle** [comment]
The related issue for fixing the PUT response is here: #1183.

**Comment by ConnorDoyle** [comment]
:shipit:, with the expanded message in the response.

**Pull request merged**